k9mail-library/src/main/java/com/fsck/k9/mail/Flag.java

```diff
@@ -56,10 +87,114 @@
      * incomplete or broken.
      * TODO Messages with this flag should be redownloaded, if possible.
      */
-    X_MIGRATED_FROM_V50,
+    public static final Flag X_MIGRATED_FROM_V50 = createInternalFlag("X_MIGRATED_FROM_V50");
 
     /**
      * This flag is used for drafts where the message should be sent as PGP/INLINE.
      */
-    X_DRAFT_OPENPGP_INLINE,
+    public static final Flag X_DRAFT_OPENPGP_INLINE = createInternalFlag("X_DRAFT_OPENPGP_INLINE");
+
+    public static final Flag X_GOT_ALL_HEADERS = createInternalFlag("X_GOT_ALL_HEADERS");
+
+
+    // Internal unique code for use in k9 database, e.g. "DELETED"
+    private final String code;
+
+    // Code for use in external storage, e.g. "\\Deleted"
+    // For IMAP: A 'system flag' or a 'keyword' in terms of RFC 3501, 2.3.2.
+    private final String externalCode;
```

Comment:
We should think about WebDAV and EWS/EAS here a bit. Ideally we want the same flags in multiple stores.